tsl/src/gapfill/exec.c

```diff
@@ -197,38 +384,71 @@ gapfill_begin(CustomScanState *node, EState *estate, int eflags)
 	List	   *targetlist = copyObject(state->csstate.ss.ps.plan->targetlist);
 	Node	   *entry;
 	bool		isnull;
-	Datum		arg1,
-				arg3,
-				arg4;
+	Datum		arg_value;
 	int			i;
 
 	state->gapfill_typid = func->funcresulttype;
 	state->state = FETCHED_NONE;
 	state->subslot = NULL;
 	state->scanslot = MakeSingleTupleTableSlot(tupledesc);
 
-	/* time_bucket interval */
-	arg1 = gapfill_exec_expr(state, linitial(func->args), &isnull);
-	/* isnull should never be true because time_bucket_gapfill is STRICT */
-	Assert(!isnull);
-	state->gapfill_period = gapfill_period_get_internal(func->funcresulttype, exprType(linitial(func->args)), arg1);
+	/* bucket_width */
+	arg_value = gapfill_exec_expr(state, linitial(func->args), &isnull);
+	if (isnull)
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("invalid bucket_width: cannot be NULL")));
+	}
+	state->gapfill_period = gapfill_period_get_internal(func->funcresulttype, exprType(linitial(func->args)), arg_value);
 
 	/*
-	 * pass gapfill start through time_bucket so it is aligned with bucket
-	 * start
+	 * check if gapfill start was left out so we have to deduce from WHERE
+	 * clause
 	 */
-	call->args = list_make2(linitial(func->args), lthird(func->args));
-	arg3 = gapfill_exec_expr(state, (Expr *) call, &isnull);
-	/* isnull should never be true because time_bucket_gapfill is STRICT */
-	Assert(!isnull);
-	state->gapfill_start = gapfill_datum_get_internal(arg3, func->funcresulttype);
+	if (IsA(lthird(func->args), Const) &&castNode(Const, lthird(func->args))->constisnull)
```

Comment:
(Just some grumbling, don't take this two seriously)

1. We're using all these list functions to access the arguments, it really feels like we should be able to write a nice `List -> struct` function to get them all at once an give them names.
2. I can't believe pgindent thinks this is the syntax to use 🤦‍♂️

Reply:
well our pgindent suffering has an ending soon :)
i wrote a static function to wrap the const + isnull check